modules/core/src/main/scala/org/tessellation/infrastructure/trust/storage/TrustStorage.scala

```diff
@@ -6,47 +6,162 @@ import cats.syntax.flatMap._
 import cats.syntax.functor._
 import cats.syntax.functorFilter._
 import cats.syntax.option._
+import cats.syntax.partialOrder._
 
 import org.tessellation.domain.trust.storage.TrustStorage
+import org.tessellation.schema.SnapshotOrdinal
 import org.tessellation.schema.peer.PeerId
-import org.tessellation.schema.trust.{PeerObservationAdjustmentUpdateBatch, PublicTrust, TrustInfo}
+import org.tessellation.schema.trust._
+
+import derevo.cats.show
+import derevo.derive
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.NonNegLong
 
 object TrustStorage {
 
-  def make[F[_]: Monad: Ref.Make]: F[TrustStorage[F]] = Ref[F].of[Map[PeerId, TrustInfo]](Map.empty).map(make(_))
+  @derive(show)
+  case class TrustMap(trust: Map[PeerId, TrustInfo], peerLabels: PublicTrustMap)
+
+  object TrustMap {
+    val empty: TrustMap = TrustMap(Map.empty, PublicTrustMap.empty)
+
+    def toPublicTrust(trust: TrustMap): PublicTrust = PublicTrust(trust.trust.mapFilter(_.publicTrust))
+  }
+
+  val snapshotInterval: NonNegLong = 1000L
+
+  val delayToUpdateCurrent: NonNegLong = 500L
+
+  private case class TrustStore(
+    trust: TrustMap,
+    currentOrdinalTrust: OrdinalTrustMap,
+    nextOrdinalTrust: OrdinalTrustMap
+  )
+
+  @derive(show)
+  case class PublicTrustMap(
+    value: Map[PeerId, PublicTrust]
+  ) {
+    def add(peerId: PeerId, publicTrust: PublicTrust): PublicTrustMap =
+      PublicTrustMap(value + (peerId -> publicTrust))
+  }
+
+  object PublicTrustMap {
+    val empty: PublicTrustMap = PublicTrustMap(Map.empty)
+  }
+
+  case class OrdinalTrustMap(
+    ordinal: SnapshotOrdinal,
+    peerLabels: PublicTrustMap,
+    trust: TrustMap
+  )
+
+  object OrdinalTrustMap {
+    def empty: OrdinalTrustMap = OrdinalTrustMap(SnapshotOrdinal.MinValue, PublicTrustMap.empty, TrustMap.empty)
+
+    def initial: OrdinalTrustMap = OrdinalTrustMap(SnapshotOrdinal(snapshotInterval), PublicTrustMap.empty, TrustMap.empty)
+
+    def advanceInterval(trust: OrdinalTrustMap): OrdinalTrustMap = OrdinalTrustMap(
+      trust.ordinal.plus(snapshotInterval),
+      PublicTrustMap.empty,
+      TrustMap.empty
```

Comment:
shouldn't it reuse the trust maps from the `trust`? what is the use case for creating an empty `OrdinalTrustMap`? if you want to create an empty case class, maybe it's better to calculate ordinal first and then create a class?

Reply:
I'm not sure I understand here.